.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,188 @@
+name: "Unit tests"
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.experimental }} # Experimental builds may fail
+
+    defaults:
+      run:
+        shell: bash -l -e -o pipefail {0}
+
+    env:
+      SINGULARITY_VERSION: "4.1.0"
+      PERL_CACHE: ~/perl5 # Perlbrew and CPAN modules installed here, cached
+      NPG_LIB: ~/perl5npg # NPG modules installed here, not cached
+      WSI_NPG_GITHUB_URL: https://github.com/wtsi-npg
+      WSI_NPG_BUILD_BRANCH: ${{ github.base_ref || github.ref }}
+
+    strategy:
+      matrix:
+        include:
+          - perl: "5.22.4"
+            irods: "4.2.7"
+            client_image: "ghcr.io/wtsi-npg/ub-16.04-irods-clients-4.2.7:latest"
+            server_image: "ghcr.io/wtsi-npg/ub-16.04-irods-4.2.7:latest"
+            experimental: false
+
+    services:
+      irods-server:
+        image: ${{ matrix.server_image }}
+        ports:
+          - "1247:1247"
+          - "20000-20199:20000-20199"
+        options: >-
+          --health-cmd "nc -z -v localhost 1247"
+          --health-start-period 30s
+          --health-interval 10s
+          --health-timeout 20s
+          --health-retries 6
+
+    steps:
+      - name: "Set environmental variables based on other environmental variables"
+        run: |
+          echo "SINGULARITY_CACHEDIR=$HOME/.singularity-cache" >> $GITHUB_ENV
+          # '~' in SINGULARITY_CACHEDIR value (from say a env: section in this YAML) is not expanded by
+          # singularity so that paths used are misleading/unclear
+
+      - uses: actions/checkout@v4
+
+      - name: "Install Singularity"
+        run: |
+          pkg="singularity-ce_${SINGULARITY_VERSION}-$(lsb_release -sc)_amd64.deb"
+          curl -sSL -O "https://github.com/sylabs/singularity/releases/download/v${SINGULARITY_VERSION}/$pkg"
+          curl -sSL -O "https://github.com/sylabs/singularity/releases/download/v${SINGULARITY_VERSION}/sha256sums"
+          grep "$os_release" sha256sums | sha256sum
+          sudo apt-get install -q -y "./$pkg"
+          
+          # Avoid the runner's loop devices being exhausted
+          sudo singularity config global --set "max loop devices" 128
```

Comment:
Assuming this is fixed in singularity 4+, do we want to keep the manual setting?

Reply:
It's probably not necessary, but I wanted a run to pass with it in place first. I'll try without

Reply:
Tried a few times with the change myself, and things were happy. But we can keep it until we are sure. Will approve the PR and you make the call when merging.